app/src/main/java/org/worldbank/transport/driver/activities/RecordFormActivity.java

```diff
@@ -292,42 +287,25 @@ private FormSectionController addSectionModel() {
                             continue;
                         }
 
-                        ArrayList<Object> enumValueObjectList = new ArrayList<>(Arrays.asList(enumClass.getEnumConstants()));
-                        ArrayList<String> enumLabels = new ArrayList<>(enumValueObjectList.size());
-
-                        for (Object enumConstant : enumValueObjectList) {
-                            String prettyLabel = enumConstant.toString();
-
-                            Enum myEnum = (Enum) enumConstant;
-                            Log.d(LOG_LABEL, "found enum #" + myEnum.ordinal() + ": " + myEnum.name());
-
-                            // Find the field of the same name as the enum constant, to get its label from
-                            // the SerializedName annotation.
-                            try {
-                                Field enumField = enumClass.getField(myEnum.name());
-                                SerializedName serializedName = enumField.getAnnotation(SerializedName.class);
-                                if (serializedName != null) {
-                                    prettyLabel = serializedName.value();
-                                }
-                            } catch (NoSuchFieldException e) {
-                                Log.e(LOG_LABEL, "Failed to find enum field to build label for " + prettyLabel);
-                                e.printStackTrace();
-                            }
-                            enumLabels.add(prettyLabel);
-                        }
+                        SelectListInfo enumListInfo = buildSelectEnumInfo(enumClass);
 
-                        Log.d(LOG_LABEL, "enumLabels: " + enumLabels.toString());
-                        Log.d(LOG_LABEL, "enumValues: " + enumValueObjectList.toString());
-
-                        // TODO: no matter what gets passed for the prompt argument, it seems to always display "Select"
-                        control = new SelectionController(this, fieldName, fieldLabel, isRequired, "Select", enumLabels, enumValueObjectList);
+                        // TODO: fix or remove prompt arg in form builder library
+                        // no matter what gets passed for the prompt argument, it seems to always display "Select"
+                        control = new SelectionController(this, fieldName, fieldLabel, isRequired, "Select",
+                                enumListInfo.labels, enumListInfo.items);
```

Comment:
If this is the IntelliJ default then probably best to leave it be, at least it'll be consistent.